src/wallet/walletdb.cpp

```diff
@@ -928,27 +928,35 @@ void NotifyBacked(const CWallet& wallet, bool fSuccess, std::string strMessage)
     wallet.NotifyWalletBacked(fSuccess, strMessage);
 }
 
-bool BackupWallet(const CWallet& wallet, const fs::path& strDest, bool fEnableCustom)
+Optional<fs::path> GetBackupPath(const CWallet& wallet)
 {
     fs::path pathCustom;
-    fs::path pathWithFile;
-    if(fEnableCustom) {
-        pathWithFile = gArgs.GetArg("-backuppath", "");
-        if(!pathWithFile.empty()) {
-            if(!pathWithFile.has_extension()) {
-                pathCustom = pathWithFile;
-                pathWithFile /= wallet.GetUniqueWalletBackupName();
-            } else {
-                pathCustom = pathWithFile.parent_path();
-            }
-            try {
-                fs::create_directories(pathCustom);
-            } catch (const fs::filesystem_error& e) {
-                NotifyBacked(wallet, false, strprintf("%s\n", e.what()));
-                pathCustom = "";
-            }
+    fs::path pathWithFile = gArgs.GetArg("-backuppath", "");
+    if(!pathWithFile.empty()) {
+        if(!pathWithFile.has_extension()) {
+            pathCustom = pathWithFile;
+        } else {
+            pathCustom = pathWithFile.parent_path();
+        }
+        try {
+            fs::create_directories(pathCustom);
+        } catch (const fs::filesystem_error& e) {
+            NotifyBacked(wallet, false, strprintf("%s\n", e.what()));
+            pathCustom = "";
+            return nullopt;
         }
     }
+    return Optional<fs::path>(pathCustom);
+}
+
+bool BackupWallet(const CWallet& wallet, const fs::path& strDest)
+{
+    Optional<fs::path> opPathCustom = GetBackupPath(wallet);
+    if (!opPathCustom) return false;
+
+    // Custom backup dir and subsequent dir/file.dat
+    fs::path pathCustom = *opPathCustom;
+    fs::path pathWithFile = pathCustom / wallet.GetUniqueWalletBackupName();
```

Comment:
We shouldn't return early when no custom path is provided, we should continue with empty `pathWithFile`.
I think we can remove the Optional and just return the (possibly empty) `fs::path` from `GetBackupPath`.

Reply:
good catch 👍 .